File: Python/SVM.py
from sklearn import datasets
from sklearn.model_selection import train_test_split
from sklearn.svm import LinearSVC

# Load 'Iris' dataset
iris = datasets.load_iris()
X = iris.data
y = iris.target

# Split dataset into training and test sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

# Create and train a linear SVM classifier
clf = LinearSVC(
    penalty='l2',
    loss='squared_hinge',
    dual=False,  # Set to False for consistency with R's primal formulation
    tol=0.001,   # Matches R's default tolerance
    C=1.0,
    multi_class='ovr',
    fit_intercept=True,
    intercept_scaling=1,
    class_weight=None,
    random_state=42,
    max_iter=1000
)
clf.fit(X_train, y_train)

File: R/SVM.R
library(e1071)
library(caret)

# Load 'Iris' dataset
data(iris)

# Split dataset into training and test sets
set.seed(42)
train_indices <- createDataPartition(iris$Species, p = 0.7, list = FALSE)
train_data <- iris[train_indices, ]
test_data <- iris[-train_indices, ]

# Create and train a linear SVM classifier
model <- svm(
    x = train_data[, -5],
    y = train_data$Species,
    type = "C-classification",
    kernel = "linear",
    cost = 1,        # Equivalent to C in sklearn
    tolerance = 0.001,  # Equivalent to tol in sklearn
    scale = FALSE,
    fitted = TRUE,
    cache.size = 40,
    epsilon = 0.1
)